lib/generators/avo/resource_generator.rb

```diff
@@ -1,9 +1,166 @@
+
+  private
+    def fields
+      fields = {}
+      fields = fields.merge(generated_params) if options[:'generate-fields']
+      fields = fields.merge(additional_params) if !additional_fields.nil?
+      fields = fields.except('id') if fields.key?('id')
+
+      fields
+    end
+
+    def generated_params
+      columns_with_type = {}
+      begin
+        model = @model_class.classify.constantize
+        model.columns_hash.each { |k, v| columns_with_type[k] = v.type }
+      rescue NameError => e
+        puts 'Name error occurs. There is no ' + @model_class.classify + ' model.'
+      rescue => e
+        puts 'Other error occured.'
+      end
+
+      result = {}
+      if !columns_with_type.empty?
+        columns_with_type.each do |fieldname, fieldtype|
+          result[fieldname] = field(fieldname, fieldtype)
+        end
+      end
+
+      result
+    end
+
+    def additional_params
+      result = {}
+
+      additional_fields.each do |fieldname, fieldtype|
+        if avo_fields.include? fieldtype
+          result[fieldname] = { field: fieldtype, }
+        else
+          result[fieldname] = field(fieldname, fieldtype)
+        end
+      end
+
+      result
+    end
+
+    def avo_fields
+      avo_fields = []
+
+      avo_fields_files = Dir[Avo::Engine.root.join('lib', 'avo', 'app', 'fields').to_s + '/*.rb']
+
+      avo_fields_files.each do |file|
+        filename = file.match('[^/]*$').to_s
+        if filename.include? 'field'
+          avo_fields.push(filename.first(-9))
+        else
+          avo_fields.push(filename.first(-3))
+        end
+      end
+
+      avo_fields.delete('')
+
+      avo_fields
+    end
+
+    def field(fieldname, fieldtype)
+      field_mappings = {
+        primary_key: {
+          field: 'id',
+        },
+        string: {
+          field: 'text',
+        },
+        text: {
+          field: 'textarea',
+        },
+        integer: {
+          field: 'number',
+        },
+        float: {
+          field: 'number',
+        },
+        decimal: {
+          field: 'number',
+        },
+        datetime: {
+          field: 'datetime',
+        },
+        timestamp: {
+          field: 'datetime',
+        },
+        time: {
+          field: 'datetime',
+        },
+        date: {
+          field: 'date',
+        },
+        binary: {
+          field: 'number',
+        },
+        boolean: {
+          field: 'boolean',
+        },
+        references: {
+          field: 'belongs_to',
+        },
+        json: {
+          field: 'code',
+        },
+      }
+
+      name_mappings = {
+        id: {
+          field: 'id',
+        },
+        description: {
+          field: 'textarea',
+        },
+        gravatar: {
+          field: 'gravatar',
+        },
+        email: {
+          field: 'text',
+        },
+        password: {
+          field: 'password',
+        },
+        password_confirmation: {
+          field: 'password',
+        },
+        stage: {
```

Comment:
IDK if we should generate this as a select. no options given.

We can use `models.defined_enums` to get all the enums ans set them as a select with `enum` option.
ex: `select :stage, enum: ::Project.stages`